regression/cbmc/struct15/test.desc

```diff
@@ -1,4 +1,4 @@
-CORE smt-backend
+CORE smt-backend new-smt-backend
 main.c
 --trace --z3
```

Comment:
Despite the `--z3` flag here, this *does* run with the new SMT backend when the `test.pl` profile for the `new-smt-backend` is run.

I have verified this locally, but I'm not sure I understand why that is - I speculate that the handling for the `--incremental-smt2-solver` flag is earlier, and path dependence handles the rest, but I haven't looked to close.

Reply:
That matches my understanding. The `--z3` flag is ignored when `--incremental-smt2-solver` is specified. Unfortunately the control flow for the `solver_factory` where this is handled is not always the easiest to follow.